include/seqan3/range/decorator/gap_decorator_anchor_set.hpp

```diff
@@ -0,0 +1,766 @@
+     * \returns Iterator to the behind last element.
+     *
+     * \attention This element acts as a placeholder; attempting to dereference it results in undefined behaviour.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    iterator end() const noexcept
+    {
+        return iterator{*this, size()};
+    }
+
+    //!\copydoc end()
+    const_iterator cend() const noexcept
+    {
+        return const_iterator{*this, size()};
+    }
+    //!\}
+
+    /*!\name Element access
+     * \{
+     */
+    /*!\brief Return the i-th element as a reference.
+     * \param i     The element to retrieve.
+     * \returns     A reference of the gapped alphabet type.
+     *
+     * ### Complexity
+     *
+     * \f$O(\log k)\f$ if \f$k\f$ is the number of gaps.
+     *
+     * ### Exceptions
+     *
+     * Throws std::out_of_range exception if \p i is out of range.
+     */
+    reference at(size_type const i)
+    {
+        if (i >= size()) // [[unlikely]]
+            throw std::out_of_range{"Trying to access element behind the last in gap_decorator."};
+        return (*this)[i];
+    }
+
+    //!\copydoc at()
+    const_reference at(size_type const i) const
+    {
+        if (i >= size()) // [[unlikely]]
+            throw std::out_of_range{"Trying to access element behind the last in gap_decorator."};
+        return (*this)[i];
+    }
+
+    /*!\brief Return the i-th element as a reference.
+     * \param i     The element to retrieve.
+     * \returns     A reference of the gapped alphabet type.
+     *
+     * This function delegates to an iterator seqan3::gap_decorator_anchor_set.
+     *
+     * ### Complexity
+     *
+     * \f$O(\log k)\f$ if \f$k\f$ is the number of gaps.
+     *
+     */
+    constexpr reference operator[](size_type const i) const noexcept
+    {
+        return *iterator{*this, i};
+    }
+    //!\}
+
+    /*!\name Comparison operators
+     * \{
+     */
+    /*!\brief Compare seqan3::gap_decorator_anchor_set by underlying sequence and gaps.
+     * \param[in] lhs The left-hand side gap decorator to compare.
+     * \param[in] rhs The right-hand side gap decorator to compare.
+     * \returns A boolean flag indicating (in)equality of the aligned sequences.
+     *
+     * ### Complexity
+     * Worst case: \f$O(n*\log k)\f$
+     * Constant in case the decorators have not the same number of (consecutive) gaps.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee. Does not modify the aligned sequences.
+     */
+    constexpr friend bool operator==(gap_decorator_anchor_set const & lhs, gap_decorator_anchor_set const & rhs) noexcept
+    {
+        if (lhs.size() != rhs.size())
+            return false;
+
+        for (size_t i = 0; i < lhs.size(); ++i)
+            if (lhs[i] != rhs[i])
+                return false;
+
+        return true;
+        // return ranges::equal(lhs, rhs); // does not work
```

Comment:
1) first compare size
2) compare equality of the std::set
3) return ranges::equal on the underlying range